### PR TITLE
fix(api-gateway): Fixes from integration testing

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Running a simple end-to-end test on a dev setup. Encountered the following bugs in the api-gateway
- Drop proxy headers before returning
- Drop Content length before proxying

### CODE REVIEW

File: src/sentry/api_gateway/proxy.py [hunk 1]
```diff
@@ -30,21 +34,30 @@ def stream_response():  # type: ignore
...
+            header_dict[header] = value
+
+
 def proxy_request(request: Request, org_slug: str) -> StreamingHttpResponse:
     """Take a django request object and proxy it to a remote location given an org_slug"""
     from sentry.types.region import get_region_for_organization
 
     target_url = get_region_for_organization(None).to_url(request.path)
-
+    header_dict = _strip_request_headers(request.headers)
```
Comment (dashed) [inline src/sentry/api_gateway/proxy.py:56]: A test case for the `Content-Length` header being stripped would be nice to have 👀

File: src/sentry/api_gateway/proxy.py [hunk 2]
```diff
@@ -14,6 +15,9 @@
 # stream 0.5 MB at a time
 PROXY_CHUNK_SIZE = 512 * 1024
 
+# List of headers to strip from a proxied request
+HEADER_STRIKE_LIST = {"Content-Length"}
```
Comment (markstory) [inline src/sentry/api_gateway/proxy.py:19]: Do you need to take off Content-Encoding as well?

**Comment by asottile-sentry** [comment]
you've rebased primary-branch history into your own -- you'll need to rebase your changes **on top** of the latest primary branch

**Comment by AniketDas-Tekky** [comment]
@asottile-sentry I think that's what I did with my latest commit. Or is it still not right?

**Comment by asottile-sentry** [comment]
> @asottile-sentry I think that's what I did with my latest commit. Or is it still not right?

still not right -- the UI is showing 500+ commits

**Review by dashed — APPROVED** [review]

**Pull request merged**